pkg/eigenState/operatorDirectedRewardSubmissions/operatorDirectedRewardSubmissions_test.go

```diff
@@ -0,0 +1,173 @@
+			}
+
+			err = model.SetupStateForBlock(blockNumber)
+			assert.Nil(t, err)
+
+			isInteresting := model.IsInterestingLog(log)
+			assert.True(t, isInteresting)
+
+			change, err := model.HandleStateChange(log)
+			assert.Nil(t, err)
+			assert.NotNil(t, change)
+
+			strategiesAndMultipliers := []struct {
+				Strategy   string
+				Multiplier string
+			}{
+				{"0x5074dfd18e9498d9e006fb8d4f3fecdc9af90a2c", "1000000000000000000"},
+				{"0xD56e4eAb23cb81f43168F9F45211Eb027b9aC7cc", "2000000000000000000"},
+			}
+
+			operatorRewards := []struct {
+				Operator string
+				Amount   string
+			}{
+				{"0x9401E5E6564DB35C0f86573a9828DF69Fc778aF1", "30000000000000000000000"},
+				{"0xF50Cba7a66b5E615587157e43286DaA7aF94009e", "40000000000000000000000"},
+			}
+
+			typedChange := change.([]*OperatorDirectedRewardSubmission)
+			assert.Equal(t, len(strategiesAndMultipliers)*len(operatorRewards), len(typedChange))
+
+			for _, submission := range typedChange {
+				assert.Equal(t, strings.ToLower("0xd36b6e5eee8311d7bffb2f3bb33301a1ab7de101"), strings.ToLower(submission.Avs))
+				assert.Equal(t, strings.ToLower("0x0ddd9dc88e638aef6a8e42d0c98aaa6a48a98d24"), strings.ToLower(submission.Token))
+				assert.Equal(t, strings.ToLower("0x7402669fb2c8a0cfe8108acb8a0070257c77ec6906ecb07d97c38e8a5ddc66a9"), strings.ToLower(submission.RewardHash))
+				assert.Equal(t, uint64(2419200), submission.Duration)
+				assert.Equal(t, int64(1725494400), submission.StartTimestamp.Unix())
+				assert.Equal(t, int64(2419200+1725494400), submission.EndTimestamp.Unix())
+
+				assert.Equal(t, strings.ToLower(strategiesAndMultipliers[submission.StrategyIndex].Strategy), strings.ToLower(submission.Strategy))
+				assert.Equal(t, strategiesAndMultipliers[submission.StrategyIndex].Multiplier, submission.Multiplier)
+
+				assert.Equal(t, strings.ToLower(operatorRewards[submission.OperatorIndex].Operator), strings.ToLower(submission.Operator))
+				assert.Equal(t, operatorRewards[submission.OperatorIndex].Amount, submission.Amount)
+			}
+
+			err = model.CommitFinalState(blockNumber)
+			assert.Nil(t, err)
+
+			rewards := make([]*OperatorDirectedRewardSubmission, 0)
+			query := `select * from operator_directed_reward_submissions where block_number = ?`
+			res := model.DB.Raw(query, blockNumber).Scan(&rewards)
+			assert.Nil(t, res.Error)
+			assert.Equal(t, len(strategiesAndMultipliers)*len(operatorRewards), len(rewards))
+
+			submissionCounter += len(strategiesAndMultipliers) * len(operatorRewards)
+
+			stateRoot, err := model.GenerateStateRoot(blockNumber)
+			assert.Nil(t, err)
+			assert.NotNil(t, stateRoot)
+			assert.True(t, len(stateRoot) > 0)
+
+			t.Cleanup(func() {
+				teardown(model)
+			})
+		})
+
+		t.Cleanup(func() {
+			teardown(model)
+		})
+	})
+
+	t.Cleanup(func() {
+		// postgres.TeardownTestDatabase(dbName, cfg, grm, l)
```

Comment:
make sure to uncomment this, otherwise it wont delete the test database and they'll start to pile up (this is likely my fault if its a copy/paste from another test file)

Reply:
Done in [54799ab](https://github.com/Layr-Labs/sidecar/pull/112/commits/54799ab82c13d42e8f5f4a550806567f833d8bd3)